src/ape/utils/misc.py

```diff
@@ -243,11 +243,15 @@ class cached_iterator(property):
     """
 
     def __init__(self, fget=None, fset=None, fdel=None, doc=None):
-        self.cache: Optional[Iterator] = None
+        self.cache: Dict[str, Iterator] = {}
         super().__init__(fget=fget, fset=fset, fdel=fdel, doc=doc)
 
     def __get__(self, obj, objtype=None):
-        iterator, self.cache = tee(self.cache if self.cache else self.fget(obj))
+        getter_id = f"{id(obj)}{id(self.fget)}"
```

Comment:
I am not sure this `@cached_iterator` is really doing much, the more I think about it...

Reply:
Maybe just remove it?